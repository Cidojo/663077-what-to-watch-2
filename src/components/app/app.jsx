import * as React from 'react';
import * as PropTypes from 'prop-types';
import {MainPage} from './../main-page/main-page.jsx';
import {MoviePage} from './../movie-page/movie-page.jsx';
import {connect} from 'react-redux';
import {ActionCreator} from './../../reducer.js';
import {movieCardPropTypes} from './../../global-custom-types';

const App = (props) => {
  const {
    genre,
    genres,
    movieCards,
    userData,
    currentVideoID,
    maxCatalogCardsOnMainPage,
    maxCatalogCardsOnDetailsPage,
    onGenreFilterClick,
    onCurrentVideoIDChange
  } = props;

  switch (location.pathname) {
    case `/`:
      return (
        <MainPage
          genre={genre}
          genres={genres}
          movieCards={movieCards}
          userData={userData}
          currentVideoID={currentVideoID}
          maxCatalogCards={maxCatalogCardsOnMainPage}
          onGenreTabClick={onGenreFilterClick}
          onCurrentVideoIDChange={onCurrentVideoIDChange}
        />
      );
    case `/details`:
      return (
        <MoviePage
          movieCards={movieCards}
          currentVideoID={currentVideoID}
          userData={userData}
          maxCatalogCards={maxCatalogCardsOnDetailsPage}
        />
      );
  }

  return null;
};

App.propTypes = {
  genre: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  movieCards: PropTypes.arrayOf(movieCardPropTypes),
  userData: PropTypes.shape({
    name: PropTypes.string,
    avatar: PropTypes.string
  }),
  currentVideoID: PropTypes.number,
  maxCatalogCardsOnMainPage: PropTypes.number,
  maxCatalogCardsOnDetailsPage: PropTypes.number,
  maxGenresToDisplay: PropTypes.number,
  onGenreFilterClick: PropTypes.func.isRequired,
  onCurrentVideoIDChange: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  genre: state.genre,
  movieCards: state.movieCards,
  currentVideoID: state.currentVideoID
});

const mapDispatchToProps = (dispatch) => ({
  onGenreFilterClick: (genre) => {
    dispatch(ActionCreator.changeGenre(genre));
  },
  onCurrentVideoIDChange: (id) => {
    dispatch(ActionCreator.changeCurrentVideoID(id));
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
