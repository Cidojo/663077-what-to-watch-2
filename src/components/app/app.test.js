import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import {App} from './app.jsx';

configure({adapter: new Adapter()});

it(`should render App component without errors`, () => {
  const tree = shallow(<App/>);
  expect(toJson(tree)).toMatchSnapshot();
});
