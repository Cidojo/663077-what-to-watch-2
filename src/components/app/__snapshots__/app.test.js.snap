// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component without errors 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPlayerScreen)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithPlayerScreen)",
          "type": [Function],
        }
      }
      exact={true}
      path="/films/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(WithAuthAccess))",
          "type": [Function],
        }
      }
      exact={true}
      path="/films/:id/review"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(WithLoginForm)",
          "type": [Function],
        }
      }
      exact={true}
      path="/login"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Connect(WithAuthAccess))",
          "type": [Function],
        }
      }
      exact={true}
      path="/mylist"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Router>
`;
